src/lib/storage/table.cpp

```diff
@@ -75,11 +80,38 @@ const std::string& Table::column_name(ColumnID column_id) const { return _column
 
 const std::string& Table::column_type(ColumnID column_id) const { return _column_types.at(column_id); }
 
-Chunk& Table::get_chunk(ChunkID chunk_id) { return _chunks.at(chunk_id); }
+Chunk& Table::get_chunk(ChunkID chunk_id) {
+  std::lock_guard<std::mutex> lock_guard(*_access_mutex);
+  return _chunks.at(chunk_id);
+}
 
 const Chunk& Table::get_chunk(ChunkID chunk_id) const { return _chunks.at(chunk_id); }
 
-void Table::compress_chunk(ChunkID chunk_id) { throw std::runtime_error("Implement Table::compress_chunk"); }
+void Table::compress_chunk(ChunkID chunk_id) {
+  std::lock_guard<std::mutex> lock_guard(*_access_mutex);
+  Chunk compressed_chunk;
+  auto& current_chunk = _chunks.at(chunk_id);
```

Comment:
Not an issue in your code as it is, but one that future modifications might cause: If someone decided to replace this with `get_chunk(chunk_id)`, you would try to recursively lock your `std::mutex`, which is undefined behavior.